samples/dialogs/dialogs.cpp

```diff
@@ -758,11 +758,13 @@ MyFrame::MyFrame(const wxString& title)
     sizer->Add(m_infoBarAdvanced, wxSizerFlags().Expand());
     SetSizer(sizer);
 
+#ifdef __WXMSW__
```

Comment:
So far I'm testing under Linux Mint, Ubuntu (default theme) and KDE.
KDE is the only one affected by this:

Before:

![dialog_not_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/2262a1f2-3759-4679-aff1-206a9791ac88)

After:
![dialog_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/e2f14d80-6f97-4f8a-9c0c-c7a26d4383da)

Reply:
Looks like the colour is being set for the entire window and not just the main widget?

Reply:
Yes, but what I don't understand is why only KDE ?
here is a screenshot of the sample under Xubuntu:

![dialog_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/00688d3f-8c5e-4859-a446-4adf920ab85e)

As you can see, the menubar and statusbar are not affected by `SetOwnBackgroundColour()`

Reply:
I don't know this neither but I still think that if SetBackgroundCopour() changes the menu bar colour, it's something that needs to be fixed in wxQt and not in the sample.

Reply:
> Looks like the colour is being set for the entire window and not just the main widget?

According to the [Qt docs](https://doc.qt.io/qt-6/application-windows.html#window-geometry) , the **menubar**, **toolbar** and **statusbar** are considered part of the client area. 

So when the background colour of the `wxFrame` is changed via `SetBackgroundColour()`, https://github.com/wxWidgets/wxWidgets/blob/99434fa005c790a10b43947d5b286a44af90847c/src/qt/window.cpp#L1263-L1272
the above controls are also affected by this call because `GetHandle()` returns `QMainWindow`. IOW, to make `SetBackgroundColour()` works like the other ports, we should use `QtGetParentWidget()` which is the same as `GetHandle()` for any window but a `wxFrame`. in which case it returns the central widget.

I tested these changes (`SetForegroundColour()` also updated) and the issue is gone.

Reply:
Great, thanks, this is exactly what I hoped was the case (sorry if I hadn't explained it clearly...) and it's great to have it fixed.

We should probably restrict other functions like this to just the central widget, if this is not already the case.

But, for now, can we remove this change from the sample and merge this already?

Thanks again!